va/validation-authority_test.go

```diff
@@ -823,6 +827,31 @@ func TestHTTPRedirectLoop(t *testing.T) {
 	fmt.Println(finChall)
 }
 
+func TestHTTPRedirectUserAgent(t *testing.T) {
+	chall := core.HTTPChallenge01(accountKey)
+	err := setChallengeToken(&chall, expectedToken)
+	test.AssertNotError(t, err, "Failed to complete HTTP challenge")
+
+	hs := httpSrv(t, expectedToken)
+	defer hs.Close()
+	port, err := getPort(hs)
+	test.AssertNotError(t, err, "failed to get test server port")
+	stats, _ := statsd.NewNoopClient()
+	va := NewValidationAuthorityImpl(&PortConfig{HTTPPort: port}, nil, stats, clock.Default())
+	va.DNSResolver = &mocks.DNSResolver{}
+	va.UserAgent = rejectUserAgent
+
+	log.Clear()
+	setChallengeToken(&chall, pathMoved)
+	finChall, _ := va.validateHTTP01(ident, chall)
+	test.AssertNotEquals(t, finChall.Status, core.StatusValid)
+
+	log.Clear()
```

Comment:
These log.Clear's are just noise here. 

Sorry, this test file is pretty rough to work with. We've not gotten around to straightening it out.